src/sieve.py

```diff
@@ -14,7 +14,10 @@ def sieve(n: int) -> list[int]:
     assert n > 0
     candidates = list(range(2, n + 1))
     primes = []
-
-    # FIXME: fill out this bit
-
+    for i in range(len(candidates)+1):
+        if len(candidates) != 0:
+            p=candidates[0]
+        else: break
+        candidates=[candidates[i] for i in range(len(candidates)) if candidates[i]%p !=0]
```

Comment:
You don't need to run through the indices and look up values here. You can just run through the values in the list.

```python
candidates = [cand for cand in candidates if cand%p != 0]
```